api.js

```diff
@@ -108,6 +109,15 @@ Api.prototype._handleExceptions = function (data) {
 };
 
 Api.prototype._handleStats = function (stats) {
+	if (this.hasExclusive && !stats.hasExclusive) {
+		return;
+	}
+
+	if (!this.hasExclusive && stats.hasExclusive) {
+		this.hasExclusive = true;
+		this.testCount = 0;
```

Comment:
Yup, same logic. Again, not too keen on it but couldn't think of a more elegant solution. Suggestions welcome!